IPython/terminal/interactiveshell.py

```diff
@@ -239,7 +239,7 @@ def prompt():
             # Ignore blank lines and consecutive duplicates
             cell = cell.rstrip()
             if cell and (cell != last_cell):
-                history.append(cell)
+                history.append(cast_unicode_py2(cell))
```

Comment:
Done. I'd still like someone to validate that this actually fixes the issue, though...